lib/authentication.rb

```diff
@@ -0,0 +1,15 @@
+require "jwt"
+
+class Authentication
+  ALGORITHM = "HS256".freeze
+
+  class << self
+    def encode payload
+      JWT.encode(payload, ENV["AUTH_SECRET"], ALGORITHM)
+    end
+
+    def decode token
+      JWT.decode(token, ENV["AUTH_SECRET"], true, algorithm: ALGORITHM).first
+    end
```

Comment:
2 thằng này hoạt động cụ thể nó như nào bạn ?

Reply:
sau khi người dùng đăng nhập sẽ trả về 1 token , token này được tạo ra bởi hàm encode. Hàm này cần sử dụng payload(chứa các thông tin mà mk muốn đặt trong token mà ở đây là user_id), khóa bí mặt và thuật toán .
khi mà muốn thao tác gì đó phaỉ gửi chuỗi token đó lên và sử dụng decode để lấy thông tin mình đặt trong đó